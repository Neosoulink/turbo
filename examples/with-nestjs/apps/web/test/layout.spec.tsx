import { render } from '@testing-library/react';

import { metadata, RootLayoutWithoutHtml } from '../app/layout';

describe('Root layout', () => {
  const { container, unmount } = render(<RootLayoutWithoutHtml />, {
    container: document.createElement('html'),
  });

  it('should match the snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  describe('metadata', () => {
    it('should be exported', () => {
      expect(metadata).toBeDefined();
    });

    it('should contain a `title` and `description`', () => {
      expect(metadata).toHaveProperty('title');
      expect(metadata).toHaveProperty('description');
    });
  });

  afterAll(() => {
    unmount();
  });
});
