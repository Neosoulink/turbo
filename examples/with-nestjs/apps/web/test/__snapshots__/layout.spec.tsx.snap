// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root layout should match the snapshot 1`] = `
<html>
  <body
    class="className"
  />
</html>
`;
